src/components/generic-filter.js

```diff
@@ -0,0 +1,251 @@
+            <Grid
+                item
+                style={{ visibility: 'hidden' }}
+                className={classes.controlItem}
+            >
+                <Switch />
+            </Grid>
+            <Grid xs={2} item className={classes.idText}>
+                <Typography component="span" variant="body1">
+                    <FormattedMessage id={'equipmentType'} />
+                </Typography>
+            </Grid>
+            <Grid xs item>
+                <Select
+                    defaultValue={type}
+                    onChange={(e) => onChange(e.target.value)}
+                    disabled={disabled}
+                    variant={'outlined'}
+                >
+                    {Object.entries(equipmentsDefinition).map(
+                        ([key, value]) => (
+                            <MenuItem key={key} value={key}>
+                                <FormattedMessage id={value.label} />
+                            </MenuItem>
+                        )
+                    )}
+                </Select>
+            </Grid>
+        </Grid>
+    );
+};
+
+export const GenericFilter = ({ initialFilter, onChange }) => {
+    const [filterType, setFilterType] = useState(initialFilter.type);
+    const currentEdit = useRef({ type: { enabled: true, value: filterType } });
+
+    const editDone = () => {
+        let res = {};
+        Object.entries(currentEdit.current).forEach(([key, value]) => {
+            if (value.enabled) res[key] = value.value;
+        });
+        onChange(res);
+    };
+
+    const changeFilterType = (newType) => {
+        currentEdit.current.type = newType;
+        setFilterType(newType);
+    };
+
+    const renderFilter = (key, definition) => {
+        if (currentEdit.current[key] === undefined)
+            currentEdit.current[key] = generateDefaultValue(
+                definition,
+                initialFilter[key]
+            );
+        return (
+            <SingleFilter
+                key={key}
+                filter={currentEdit.current[key]}
+                definition={definition}
+                onChange={editDone}
+            />
+        );
+    };
+
+    const RenderGeneric = () => {
+        return Object.entries(genericFields).map(([key, definition]) =>
+            renderFilter(key, definition)
+        );
+    };
+
+    const renderSpecific = () => {
+        return Object.entries(equipmentsDefinition[filterType].fields).map(
+            ([key, definition]) => {
+                if (definition.occurs)
+                    return (
+                        <Grid
+                            container
+                            item
+                            direction={'row'}
+                            key={key}
+                            spacing={1}
+                        >
+                            {[
+                                ...Array.from(Array(definition.occurs).keys()),
+                            ].map((n) => {
```

Comment:
The problem with this way to generate filters components is that it creates multiple filters with exactly the same label, and thus, we don't exactly know what the filter is used for (i.e. we can have two "countries" filters, and we don't exactly know what they really do)